synapse/lib/certdir.py

```diff
@@ -15,14 +15,74 @@
 if defdir is None:
     defdir = defdir_default
 
+logger = logging.getLogger(__name__)
+
+TEN_YEARS = 10 * 365 * 24 * 60 * 60
+
+
 def iterFqdnUp(fqdn):
     levs = fqdn.split('.')
     for i in range(len(levs)):
         yield '.'.join(levs[i:])
 
-TEN_YEARS = 10 * 365 * 24 * 60 * 60
+TLS_SERVER_CIPHERS = ''
 
-logger = logging.getLogger(__name__)
+def initTLSServerCiphers():
+    '''
+    Create a cipher string that supports TLS 1.2 and TLS 1.3 ciphers which do not use RSA.
+
+    Note:
+        The results of this may be dynamic depending on the interpreter version and OpenSSL library in use.
+        For Python 3.8 and below, the cipher list is a subset of the normal default ciphers which commonly available.
+        For Python 3.10+, the changes should be negligible.
+
+        The resulting string is cached in the module global TLS_SERVER_CIPHERS and called at import time.
+
+    Returns:
+        str: A OpenSSL Cipher string.
+    '''
+
+    global TLS_SERVER_CIPHERS
```

Comment:
A simple `TLS_SERVER_CIPHERS = _initTLSServerCiphers()` would be tighter than using globals and implementing re-run safety which will never be used. Make coverage a bit easier as well.